libcnb-cargo/src/main.rs

```diff
@@ -173,15 +173,20 @@ fn handle_libcnb_package(args: PackageArgs) {
         std::process::exit(exit_code::UNSPECIFIED_ERROR);
     };
 
+    let size_in_bytes = calculate_dir_size(&output_path).unwrap_or_else(|io_error| {
+        error!("IO error while calculating buildpack directory size: {io_error}");
+        std::process::exit(exit_code::UNSPECIFIED_ERROR);
+    });
+
+    // Precision will only be lost for sizes bigger than 52 bits (~4 Petabytes), and even
+    // then will only result in a less precise figure, so is not an issue.
+    #[allow(clippy::cast_precision_loss)]
+    let size_in_mb = size_in_bytes as f64 / (1024.0 * 1024.0);
```

Comment:
(Optional) Ultra pedantic mode: ENGAGED! 😂 

> In addition, the conversion of the size from bytes to MB was incorrect, since it used SI units (ie: divided by 1000\*1000 rather than 1024\*1024).

Technically, it was correct before since `MB` refers to the SI value where a megabyte is exactly a million bytes. The value here should be called a Mebibyte and using `MiB` as the abbreviation.

Reply:
Ah true. Though before it just said `M` not `MB` so from an end user perspective it wasn't displaying what we wanted it to say either way. I'll change it to `MiB` though :-)